uds.js

```diff
@@ -194,6 +194,14 @@
         var url = udsHostName.clone().setPath('/case/' + caseNumber + "/comments/public");
         return executeUdsAjaxCallWithData(onSuccess,onFailure,url,caseComment,'POST');
     };
+
+    uds.postPublicCommentsWithHours = function (onSuccess, onFailure, caseNumber, caseComment, hoursWorked) {
```

Comment:
I think instead add condition inside postPublicComments method only no need of creating a separate method:
Something like - -
if(hoursWorked == undefined) {
call '/case/' + caseNumber + "/comments/public" }
else {
call '/case/' + caseNumber + "/comments/public/hoursWorked/"+hoursWorked }